pandas/core/arrays/datetimelike.py

```diff
@@ -520,7 +520,9 @@ def __getitem__(self, key):
         if com.is_bool_indexer(key):
             # first convert to boolean, because check_array_indexer doesn't
             # allow object dtype
-            key = np.asarray(key, dtype=bool)
+            if is_object_dtype(key):
```

Comment:
Does `is_bool_indexer` imply key is an array, or is it just a list at this point? Ideally we'd do `is_object_dtype(key.dtype)`

Reply:
`key` could potentially be a list here

Reply:
But if it is a list, it is fine to pass it to `check_array_indexer` I think (so something like `[True, False, pd.NA]` works)?

Reply:
We can pass a list to `check_array_indexer`, but I think the specific case we have to worry about here is when `key` is a numpy array of object dtype, but nonetheless contains bools. In that case it passes `is_bool_indexer` but raises inside `check_array_indexer`, so we explicitly cast it to `bool` dtype here